rules/import-style.js

```diff
@@ -127,7 +128,48 @@ const defaultStyles = {
 	},
 };
 
-/** @param {import('eslint').Rule.RuleContext} context */
+const specialCases = {
+	react: 'React',
+	'react-dom': 'ReactDOM',
+	'react-router': 'ReactRouter',
+	'react-router-dom': 'ReactRouterDOM',
+	'prop-types': 'PropTypes',
+	lodash: '_',
+	'lodash-es': '_',
+	jquery: '$',
+	'styled-components': 'styled',
+	redux: 'Redux',
+	'react-redux': 'ReactRedux',
+	axios: 'Axios',
+	moment: 'moment',
+	'date-fns': 'dateFns',
+	ramda: 'R',
+	rxjs: 'Rx',
+	vue: 'Vue',
+	angular: 'Angular',
+};
+
+const getNamespaceIdentifier = moduleName => {
+	// Handle special cases first
+	if (specialCases[moduleName]) {
+		return specialCases[moduleName];
+	}
+
+	// Get the last part of the path and remove extension
+	const lastPart = moduleName.split('/').pop().split('.')[0];
+
+	// For scoped packages, we want the package name, not the scope
+	// @scope/package -> package
+	// @scope/foo.bar -> foo
+	const packageName = lastPart.startsWith('@') ? moduleName.split('/').pop().split('.')[0] : lastPart;
+
+	// Replace invalid identifier characters and convert to camelCase
+	return packageName
+		.replaceAll(/[^\dA-Za-z-]/g, '-') // Replace invalid chars with hyphen
```

Comment:
Will not be valid identifier if filename starts With number.

Reply:
Nice catch. I've updated the implementation and added test cases.